graylog2-web-interface/src/views/stores/ViewStore.ts

```diff
@@ -157,7 +157,7 @@ export const ViewStore: ViewStoreType = singletonStore(
       /* Select selected query (activeQuery) or first query in view (for now).
          Selected query might become a property on the view later. */
       const queries = view.state.keySeq().toList();
-      const firstQueryId = queries.first();
+      const firstQueryId = view?.search?.queries?.first()?.id ?? queries.first();
```

Comment:
I implemented `queries.first()` as a fallback to ensure we are definitely not breaking anything.